x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java

```diff
@@ -35,6 +35,11 @@
  * A container for xpack setting constants.
  */
 public class XPackSettings {
+
+    public static final boolean USER_PROFILE_FEATURE_FLAG_ENABLED = "true".equals(
+        System.getProperty("es.user_profile_feature_flag_enabled")
+    );
```

Comment:
I think we want to enable profiles automatically in snapshot builds so that people doing downstream testing don't need to mess with system properties.
```suggestion
    public static final boolean USER_PROFILE_FEATURE_FLAG_ENABLED = Build.CURRENT.isSnapshot() || "true".equals(
        System.getProperty("es.user_profile_feature_flag_enabled")
    );
```

Reply:
TIL. Thanks! I applied the suggestion.

PS: I was going to remove the explicit configuration for this system property in tests. But the release tests may still need them (tried test locally but stuck at downloading ml dependency). Anyway, it's no harm to have them.

Reply:
Yeah, you need to keep them, or disable the tests in release builds (which is worse).